public/general_components/multi_select_filter.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFilterSelectItem,
+  EuiFilterGroup,
+  EuiPopover,
+  EuiFilterButton,
+  EuiFlexItem,
+} from '@elastic/eui';
+
+// styling
+import './general-component-styles.scss';
+
+interface MultiSelectFilterProps {
+  title: string;
+  filters: EuiFilterSelectItem[];
+  setSelectedFilters: (filters: EuiFilterSelectItem[]) => void;
+}
+
+/**
+ * A general multi-select filter.
+ */
+export function MultiSelectFilter(props: MultiSelectFilterProps) {
+  const [filters, setFilters] = useState(props.filters);
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  function onButtonClick() {
+    setIsPopoverOpen(!isPopoverOpen);
+  }
+  function onPopoverClose() {
+    setIsPopoverOpen(false);
+  }
+
+  function updateFilter(index: number) {
+    if (!filters[index]) {
+      return;
+    }
+    const newFilters = [...filters];
+    // @ts-ignore
```

Comment:
TLDR of adding all of these ignores is due to IDE complaints that the type fields don't match. This is due to differences in EUI vs. OUI and their respective documentation. This follows the latest OUI documentation.